### ISSUE TITLE
bug: Trying to enter Login screen twice crashes the app

### ISSUE THREAD

**Issue by haroldadmin** [fossasia/open-event-android#1242]
**Describe the bug**
Trying to enter Login screen twice crashes the app

**To Reproduce**
1. Go to Login page
2. Tap on Login in Bottom Nav again
3. App crashes

**Expected behavior**
App does not crash

**Screenshots**
![crash](https://user-images.githubusercontent.com/24315306/53960695-f48ff980-410c-11e9-811f-9dfcaf2c7e7d.gif)

**Crash log**
```
2019-03-07 19:27:46.601 31803-31803/com.eventyay.attendee E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.eventyay.attendee, PID: 31803
    java.lang.IllegalStateException: Fragment LoginFragment{3bfb136 (8835c986-94ef-4728-ac33-e6a256da19f0) id=0x7f0800f4} has null arguments
        at org.fossasia.openevent.general.auth.LoginFragment$$special$$inlined$navArgs$1.invoke(FragmentNavArgsLazy.kt:42)
        at org.fossasia.openevent.general.auth.LoginFragment$$special$$inlined$navArgs$1.invoke(Unknown Source:0)
        at androidx.navigation.NavArgsLazy.getValue(NavArgsLazy.kt:44)
        at androidx.navigation.NavArgsLazy.getValue(NavArgsLazy.kt:34)
        at org.fossasia.openevent.general.auth.LoginFragment.getSafeArgs(Unknown Source:28)
        at org.fossasia.openevent.general.auth.LoginFragment.showSnackbar(LoginFragment.kt:214)
        at org.fossasia.openevent.general.auth.LoginFragment.onCreateView(LoginFragment.kt:60)
        at androidx.fragment.app.Fragment.performCreateView(Fragment.java:2530)
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:887)
        at androidx.fragment.app.FragmentManagerImpl.addAddedFragments(FragmentManagerImpl.java:2090)
        at androidx.fragment.app.FragmentManagerImpl.executeOpsTogether(FragmentManagerImpl.java:1864)
        at androidx.fragment.app.FragmentManagerImpl.removeRedundantOperationsAndExecute(FragmentManagerImpl.java:1820)
        at androidx.fragment.app.FragmentManagerImpl.execPendingActions(FragmentManagerImpl.java:1721)
        at androidx.fragment.app.FragmentManagerImpl$1.run(FragmentManagerImpl.java:147)
        at android.os.Handler.handleCallback(Handler.java:873)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6718)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
```

**Smartphone Info:**

Device: Google Pixel
OS: Android 9 Pie

**Would you like to work on the issue?**
Yes

**Comment by ShridharGoel** [comment]
Not able to reproduce the issue. Can you attach logs?

**Comment by haroldadmin** [comment]
@ShridharGoel I have updated the issue with crash logs.

The problem occurs because of Navigation AAC:
1. It tries to replace the existing fragment with the one selected in bottom nav, even if they are the same fragment.
2. When it instantiates this same fragment again, it does not pass the arguments given to the previous instance to this new instance.

I am going to work on a fix for this and send a PR soon.

**Comment by ShridharGoel** [comment]
Yes, reproduced after pulling the latest commits. Go ahead.

**Comment by theLimitBreaker** [comment]
@haroldadmin, uhh there is No PR till now,  I am almost done with this issue . Could  I proceed ?

**Comment by haroldadmin** [comment]
Um sure. My exams are going on till the end of the week, you can take up the issue if you want. What approach have you used to solve this, btw?

**Comment by theLimitBreaker** [comment]
safeArgs currently throw Execption when are null , 
But in the nav_graph.xml , in fragment -> in argument , default value is given null , app type is also set null . This is what throwing exception .

**Comment by haroldadmin** [comment]
I was thinking the exception is thrown when the fragment is instantiated without any arguments, such as when pressing the login tab in the bottom nav view. I might be wrong, though. Cool, send your PR and I'll help review it if I get the chance.

**Comment by theLimitBreaker** [comment]
By NavLazyArgs , now setting arguments comes under the control of it. Setting null to it causes Throw execption.

```
@MainThread
inline fun <reified Args : NavArgs> Fragment.navArgs() = NavArgsLazy(Args::class) {
    arguments ?: throw IllegalStateException("Fragment $this has null arguments")
}
```

Thats from its code

**Comment by haroldadmin** [comment]
You'd notice that it checks if the `arguments` property on the receiver (`Fragment`) is null or not. If it is not null, it parses it otherwise it throws an exception. This means that it doesn't care if the value of the variables inside the bundle is null. It only cares if the bundle itself is null. I maybe completely off-base here, though.

**Comment by theLimitBreaker** [comment]
You are right about arguments , but if you set default value of argument **null** then safeArgs Param itself become null . Thus throws Exception

**Comment by haroldadmin** [comment]
Oh. Okay. That makes sense.

**Comment by haroldadmin** [comment]
That makes sense. Thanks!

**Comment by theLimitBreaker** [comment]
@haroldadmin I have made PR, you could review it now :)

**Comment by haroldadmin** [comment]
The underlying issue with the Nav AAC has been fixed at the library level in the latest alpha.

https://issuetracker.google.com/issues/128531879

**Comment by theLimitBreaker** [comment]
Ok so let me update it to latest alpha . Thanks for telling :)

**Comment by haroldadmin** [comment]
If we update to the latest alpha, then there's no need to do anything regarding this issue, and the PR becomes irrelevant.

@iamareebjamal Would updating to the latest alpha be an acceptable solution?

**Comment by iamareebjamal** [comment]
Yes

**Comment by theLimitBreaker** [comment]
> The underlying issue with the Nav AAC has been fixed at the library level in the latest alpha.
> 
> https://issuetracker.google.com/issues/128531879
> 
> > **[Sign in - Google Accounts](https://issuetracker.google.com/issues/128531879)**

@haroldadmin 
This from the release notes of Navigation component

`Your project must have migrated to AndroidX to use 2.X releases of Navigation. Navigation 1.0 stable will be the last release using the Support Library dependencies; all future development beyond 1.0 will be based on AndroidX and build upon the 2.0 stable release.`

Heres the link
https://developer.android.com/jetpack/androidx/releases/navigation#2.0.0-rc02

Should I refractor to AndroidX then and update the library ,

**Comment by iamareebjamal** [comment]
AFAIK, the project is already using AndroidX

**Comment by theLimitBreaker** [comment]
Navigation component is using Pre-AndroidX artifacts

**Comment by haroldadmin** [comment]
Or we could just implement the earlier solution for now, and use the Nav component 2.0 when it is released, right @iamareebjamal?

**Comment by theLimitBreaker** [comment]
@haroldadmin 2.0 is released but 2.1.0-alpha01 is still not released(2.1.0-alpha01 is version which fixed the problem in Navigation Arch component)
